src/server/src/controllers/post.controllers.ts

```diff
@@ -48,7 +48,7 @@ const createPost = async (req: AuthenticatedRequest, res: Response) => {
         image: Buffer.from(req.file.buffer).toString('base64'),
       }),
       content: content,
-      categories: categories ? categories : [],
+      categories: categories ? JSON.parse(categories) : [],
```

Comment:
Adding works! But you forgot to parse in UpdatePost too as editing still errors.

Reply:
Good catch!